### PR TITLE
For #7363 fix flaky verifySelectTextTest UI test

### LINKED ISSUE #7363: Intermittent UI test failure - < WebControlsTest. verifySelectTextTest >
### Stacktrace:
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:87)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertTrue(Assert.java:53)
	at org.mozilla.focus.activity.robots.BrowserRobot.verifyPageContent(BrowserRobot.kt:55)
	at org.mozilla.focus.activity.WebControlsTest$verifySelectTextTest$2.invoke(WebControlsTest.kt:153)
	at org.mozilla.focus.activity.WebControlsTest$verifySelectTextTest$2.invoke(WebControlsTest.kt:148)
	at org.mozilla.focus.activity.robots.SearchRobot$Transition.loadPage(SearchRobot.kt:97)
	at org.mozilla.focus.activity.WebControlsTest.verifySelectTextTest(WebControlsTest.kt:148)

### PR DESCRIPTION
**Pull request by AndiAJ**
For #7363 fix flaky `verifySelectTextTest` UI test ✅ successfully ran 100x on Firebase

Summary: 
The problem was that the "Paste" context option sometimes wasn't displayed, as well as the fact that it wasn't properly identified, failing to click it.

Refactored in the same fashion the long click and copy of text.

Also added `NullPointerException` to the retry rule in case the click fails on a UiObject2

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on `Show All Checks`,
2. click `Details` next to `build-focus-debug` or `build-klar-debug` for changes targeting Klar,
2. click `View task in Taskcluster`,
3. click the `Artifacts` row,
4. click to download any of the apks listed here which use an appropriate name for each CPU architecture.

Fixes #7363
Fixes #7363
Fixes #7363
Fixes #7363
Fixes #7363
Fixes #7363

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 1]
```diff
@@ -321,13 +323,18 @@ class BrowserRobot {
         var currentTries = 0
         while (currentTries++ < 3) {
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:324]: I think it looks ok. But would you mind refactoring this while loop in the same style we used lately with "for (i in 0..RETRY_COUNT)"?

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 2]
```diff
@@ -342,11 +349,19 @@ class BrowserRobot {
...
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:350]: same comment as above.

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:324]: Sure, no problem !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:350]: Sure, no problem !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt]: Done !

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 3]
```diff
@@ -339,14 +349,25 @@ class BrowserRobot {
...
-        var currentTries = 0
-        while (currentTries++ < 3) {
+        for (i in 1..RETRY_COUNT) {
             try {
-                mDevice.findObject(UiSelector().textContains("Paste")).waitForExists(waitingTime)
-                mDevice.findObject(By.textContains("Paste")).also { it.click() }
+                longPressTextInputBox()
+
+                mDevice.wait(Until.findObject(By.textContains("Paste")), waitingTime)
+                Assert.assertNotNull(mDevice.findObject(By.textContains("Paste")))
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:357]: same as above.

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 4]
```diff
@@ -318,19 +320,27 @@ class BrowserRobot {
...
+        for (i in 1..RETRY_COUNT) {
             try {
-                mDevice.findObject(UiSelector().textContains(expectedText)).waitForExists(waitingTime)
-                mDevice.findObject(By.textContains(expectedText)).also { it.longClick() }
+                mDevice.wait(Until.findObject(By.textContains(expectedText)), waitingTime)
+                Assert.assertNotNull(mDevice.findObject(By.textContains(expectedText)))
+                mDevice.findObject(By.textContains(expectedText)).longClick()
+
+                mDevice.wait(Until.findObject(By.textContains("Copy")), waitingTime)
+                Assert.assertNotNull(mDevice.findObject(By.textContains("Copy")))
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:330]: same as above

File: app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt [hunk 5]
```diff
@@ -318,19 +320,27 @@ class BrowserRobot {
...
 
     fun longClickAndCopyText(expectedText: String) {
-        var currentTries = 0
-        while (currentTries++ < 3) {
+        for (i in 1..RETRY_COUNT) {
             try {
-                mDevice.findObject(UiSelector().textContains(expectedText)).waitForExists(waitingTime)
-                mDevice.findObject(By.textContains(expectedText)).also { it.longClick() }
+                mDevice.wait(Until.findObject(By.textContains(expectedText)), waitingTime)
+                Assert.assertNotNull(mDevice.findObject(By.textContains(expectedText)))
```
Comment (sv-ohorvath) [inline app/src/androidTest/java/org/mozilla/focus/activity/robots/BrowserRobot.kt:326]: could these 2 lines (wait and assertNotNull) be replaced by a single-line waitNotNull? If needed, you can increase the waitingTime to waitingTimeLong.

**Review by sv-ohorvath — APPROVED** [review]
Just some small observations, otherwise it looks ok.

**Comment by AndiAJ** [comment]
Tried with your suggestion @sv-ohorvath but it always fails due to the ANR.
Reverted back to the original implementation, re-ran the test 50x on Firebase, and it always fails again due to the ANR.
Seems to be happening due to #7449

**Comment by AndiAJ** [comment]
Performed a check and `verifyDismissTextSelectionToolbarTest` is also failing  due to the ANR.

**Comment by AndiAJ** [comment]
Closing this PR as I can't do more until the ANR is fixed.

**Pull request closed without merging**